scripts/commands/gettp.lua

```diff
@@ -0,0 +1,48 @@
+-----------------------------------
+-- func: gettp <player>
+-- desc: Displays target's current TP
+-----------------------------------
+
+cmdprops =
+{
+    permission = 1,
+    parameters = "is"
+}
+
+function error(player, msg)
+    player:PrintToPlayer(msg)
+    player:PrintToPlayer("!gettp (player)")
+end
+
+function onTrigger(player, tp, target)
+    -- validate target
+    local targ
+    local cursorTarget = player:getCursorTarget()
+
+    if target then
+        targ = GetPlayerByName(target)
+        if not targ then
+            error(player, string.format("Player named '%s' not found!", target))
+            return
+        end
+    elseif cursorTarget and not cursorTarget:isNPC() then
+        targ = cursorTarget
+    else
+        targ = player
+    end
+
+    -- get tp
+    if targ:isAlive() then
+        targ:getTP(tp)
+        local pet = targ:getPet()
+        if pet and pet:isAlive() then
+            pet:getTP(tp)
```

Comment:
Dead code?

Reply:
there is a lot of excess code here not really needed. As long as target isn't nil and isn't an NPC, just print the name and its current TP. No reason to care what type of entity it is like pet - other than npc, because npc's don't have stats like TP. Also doesn't matter if its dead when the command executes, that won't break the world.

`getTP()`doesn't have an input parameter btw.